### PR TITLE
Adding support for 'origen rc tag' without prepending 'v' to the tag

### PR DESCRIPTION
**Pull request by info-rchitect**
The change in the plugins file was a workaround so I could test out the PR.  **Question:** Would we ever allow someone to make a release where the session mode is set to 'debug'?

### CODE REVIEW

**Comment by info-rchitect** [comment]
Works internally, not sure about at other companies

File: lib/origen/application/plugins.rb [hunk 1]
```diff
@@ -14,15 +14,17 @@ def initialize
...
-        if Origen.mode.production? || force
-          if File.exist?("#{Origen.root}/Gemfile")
-            File.readlines("#{Origen.root}/Gemfile").each do |line|
-              # http://rubular.com/r/yNGDGB6M2r
-              if line =~ /^\s*gem\s+(("|')\w+("|')),.*(:path\s*=>|path:)/
-                fail "The following gem is defined as a path in your Gemfile, but that is not allowed in production: #{Regexp.last_match[1]}"
-              end
-              if line =~ /ORIGEN PLUGIN AUTO-GENERATED/
-                fail 'Fetched gems are currently being used in your Gemfile, but that is not allowed in production!'
+        unless Origen.app.session.origen_core[:mode] == 'debug'
```
Comment (ginty) [inline lib/origen/application/plugins.rb]: Don't think we can allow this change, that will seriously change behavior for existing apps

File: lib/origen/version_string.rb [hunk 2]
```diff
@@ -75,7 +75,9 @@ def valid?(_options = {})
     # Returns true if the version is a semantic format version number
     def semantic?
       !!(self =~ /^v?\d+\.\d+\.\d+$/ ||
-         self =~ /^v?\d+\.\d+\.\d+\.(dev|pre)\d+$/
+         self =~ /^v?\d+\.\d+\.\d+\.(dev|pre)\d+$/ ||
+         self =~ /^\d+\.\d+\.\d+$/ ||
```
Comment (ginty) [inline lib/origen/version_string.rb]: The v is already optional above - v?, instead of just v.
I think all the change in this file are un-necessary.

Comment (info-rchitect) [reply lib/origen/version_string.rb]: OK

File: lib/origen/application/plugins.rb (hunk 1 above)
Comment (ginty) [inline lib/origen/application/plugins.rb]: Don't like this, you can either force or you can't

**Comment by ginty** [comment]
> Question: Would we ever allow someone to make a release where the session mode is set to 'debug'?

Yes, I think we do that all the time and it would be a big breaking change to enforce that

File: lib/origen/application/plugins.rb (hunk 1 above)
Comment (info-rchitect) [reply lib/origen/application/plugins.rb]: yeah wasn't sure, where is the force initiated?

**Comment by info-rchitect** [comment]
@ginty OK I think I fixed all of your issues, pls confirm.  thx for the quick review.

**Pull request merged**